recipe/gen_patch_json.py

```diff
@@ -1706,6 +1706,26 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
                 i = record["depends"].index("pyyaml")
                 record["depends"][i] = "pyyaml >=6.0"
 
+        # Notebook 6.5.5 pins the versions of jupyter_client to < 8
+        # and pyzmq>=25 as a temporary fix for issue
+        # https://github.com/jupyter/notebook/issues/4439
+        # all versions of Notebook < 7 should at least pin to 
+        # jupyter_client < 8 and pyzmq < 25 as in
+        # https://github.com/jupyter/notebook/pull/6749
+        if record_name == "notebook" and int(record["version"].split(".", 1)[0]) < 7:
```

Comment:
Note the second checklist item in the PR template...

> Make sure to add a condition `and record.get("timestamp", 0) < NOW` so your changes only affect packages built in the past. Replace NOW with `python -c "import time; print(f'{time.time():.0f}000')"` -->

...so perhaps something like:

```py
if (
    record_name == "notebook" 
    and int(record["version"].split(".", 1)[0]) < 7
    and record.get("timestamp", 0) < 1691035398000
):
```